### PR TITLE
Support for FunAsSeq

### LINKED ISSUE #569: [FEATURE] Translate FunAsSeq in the model checker
Following #223, we have to properly translate `Apalache!FunAsSeq` to SMT.

### PR DESCRIPTION
**Pull request by konnov**
Closes #569. Instead of translating `FunAsSeq` to SMT, we are simply using its TLA+ definition via `MkSeq`.

- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] Documentation added for any new functionality
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

File: tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/ApalacheOper.scala [hunk 1]
```diff
@@ -102,10 +102,15 @@ object ApalacheOper {
...
-   * Attempt to dynamically cast an Int -> T function to a Seq(T). The first argument should be the function expression
-   * and the second argument should be an integer, denoting the maximal length of the sequence.
+   * <p>Attempt to dynamically cast an Int -> T function to a Seq(T). The first argument should be the function
+   * expression and the second argument should be an integer, denoting the maximal length of the sequence.</p>
+   *
+   * <p>We keep this operator in the IR. However, we are using the definition of this operator from Apalache.tla. Hence,
+   * if you construct an expression that contains `funAsSeq`, its constructor will throw.</p>
    */
   object funAsSeq extends ApalacheOper {
+    require(false, "This operator is defined in Apalache.tla. Do not construct it.")
```
Comment (Kukovec) [inline tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/ApalacheOper.scala:112]: Why not just delete the object, if you're going to add `require(false)`?

**Review by Kukovec — APPROVED** [review]
Looks fine, but I wonder if we couldn't change the interface of `FunAsSeq(fn, maxSeqLen)` to `FunAsSeq(fn)` instead, and just compute `maxSeqLen == FoldSet(MaxOf2, 0, DOMAIN fn)` (at the cost of performance)

File: tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/ApalacheOper.scala (hunk 1 above)
Comment (konnov) [reply tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/ApalacheOper.scala:112]: I also wanted to delete the object. But it is better to keep it as documentation.

**Comment by konnov** [comment]
> Looks fine, but I wonder if we couldn't change the interface of `FunAsSeq(fn, maxSeqLen)` to `FunAsSeq(fn)` instead, and just compute `maxSeqLen == FoldSet(MaxOf2, 0, DOMAIN fn)` (at the cost of performance)

That would not help us, as `MkSeq` requires a constant length.

**Comment by konnov** [comment]
However, we should think further. Maybe there is a way to implement `FunAsSeq` without passing constant length directly.

**Pull request merged**